spyder/utils/programs.py

```diff
@@ -710,7 +712,27 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                 else:
                     run_program(program['cmd'], arglist)
                 return
-        # TODO: Add a fallback to OSX
+    elif sys.platform == 'darwin':
+        f = tempfile.NamedTemporaryFile('wt', suffix='.sh',
+                                        dir=get_temp_dir(), delete=False)
+        f.write(' '.join(p_args))
+        f.close()
+        os.chmod(f.name, 0o777)
+
+        def run_terminal_thread():
+            if wdir:
+                proc = subprocess.Popen(
+                    'open -a Terminal.app ' + f.name, cwd=wdir, shell=True)
+            else:
+                proc = subprocess.Popen(
+                    'open -a Terminal.app ' + f.name, shell=True)
+            # Prevent race condition
+            time.sleep(3)

```

Comment:
Why do we have to wait for so long here? Could we reduce this time?

Problem is if an operation takes more than 300ms users will think something is working incorrectly or not working at all (that's something backed up by research).

Reply:
This time is necessary to prevent a race condition from the call of the subprocess and the removal of the temporal file.